core/engine_test.go

```diff
@@ -957,9 +957,9 @@ func TestMinIterationDuration(t *testing.T) {
 		require.False(t, engine.IsTainted())
 	}
 
-	// Only 2 full iterations are expected to be completed due to the 1 second minIterationDuration
-	assert.Equal(t, 2.0, getMetricSum(collector, metrics.Iterations.Name))
+	// 4 full iterations are expected to be completed
+	assert.Equal(t, 4.0, getMetricSum(collector, metrics.Iterations.Name))
 
-	// But we expect the custom counter to be added to 4 times
+	// The custom counter should be added to 4 times
```

Comment:
I feel icky modifying the test to match the result, but according to [the description](https://github.com/loadimpact/k6/issues/1189#issue-503381630):

> The only difference for local execution this should make is that currently, iterations is emitted after any VU sleep caused by minIterationDuration, and after this change it would be emitted before it.

So this is an acceptable change?

Reply:
I would argue that you should lower the `minIterationDuration` to .. `150ms`, the `duration` to `500ms` and increase the VUs to `4` and expect that you will 16 iterations, instead of a couple hundred ;)

Reply:
@MStoykov hhmm can you explain why that would be an improvement, besides taking less time overall?

To me it seems that 2 VUs registering 4 iterations in 2s is functionally equivalent to 4 VUs registering 16 iterations in 600ms.

The issue introduced in this PR is that it changes when `iterations` is emitted in regards to when the `minIterationDuration` sleep is done, as @na-- mentioned in the description.

To me this is a backwards breaking change and should be fixed before merging this, but I'm unsure on a) if you guys agree, and b) if a simple fix is possible without major changes.

Reply:
IMO the previous behaviour was ... wrong ... the iteration finished. The `minIterationDuration` is ... somewhat poorly named, I think that `minIterationTime` should've been better, but I guess it's good enough and definetely not something that I think is worth renaming, especially given the fact I don't have a much better suggestion.

 So the new breaking change behaviours is IMO the correct one ;). Also we discussed that this will happen and IFAIK agreed that it's unlikely anyone has a use case where it will matter ;) 

The functional difference (apart from the faster tests) is that the test will get through more than one `minIterationDuration` sleep, potentially catching bugs where (as until the now expected iteration) we only go through 1 iteration or we have a bug where `minIterationDuration` sleep takes WAY longer than expected ... I am not saying that it will be 10x better tests, but will definitely have better chance of catching something ;)

Reply:
Thanks for explaining, it does make the test more robust.